h2/src/main/org/h2/value/ValueTimestampTimeZone.java

```diff
@@ -50,7 +50,7 @@
      */
     private final long timeNanos;
     /**
-     * Time zone offset from UTC in minutes, range of -12hours to +12hours
+     * Time zone offset from UTC in minutes, range of -18 hours to +18 hours
```

Comment:
why? this introduces the possibility of wraparound, which means that we'd have to think carefully about what to do with the equals() method.

Reply:
Because in real world some countries have time zone offsets larger that 12h.

equals() should handle them normally.

Reply:
-12 to +14 is the largest range I can find ?
    https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

Reply:
Except for historical ones that may be useful for somebody to store old date and time values. Unlikely, of course.

From my point of view its better to use the same limit as Java have. JSR-310 was discussed by a larger group and probably decisions it it are reasonable enough.

Reply:
ah, now that is useful information! excellent! 

please add a comment to that effect here.

Reply:
I wrote a comment just before range check and also mention JSR-310 in javadoc of this field.